modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -256,6 +264,29 @@ inline void preprocess(const cv::Mat& src,
     }
 }
 
+void preprocess(const cv::MediaFrame::View& view,
+                const cv::GFrameDesc& desc,
+                const cv::gimpl::onnx::TensorInfo& ti,
+                const cv::Rect& roi,
+                      cv::Mat& dst) {
+    cv::Mat pp;
+    switch (desc.fmt) {
+        case cv::MediaFormat::BGR: {
+            pp = cv::Mat(desc.size, CV_8UC3, view.ptr[0], view.stride[0]);
+            break;
+        }
+        case cv::MediaFormat::NV12: {
+            const auto y_plane  = cv::Mat(desc.size, CV_8UC1, view.ptr[0], view.stride[0]);
+            const auto uv_plane = cv::Mat(desc.size / 2, CV_8UC2, view.ptr[1], view.stride[1]);
+            cvtColorTwoPlane(y_plane, uv_plane, pp, cv::COLOR_YUV2BGR_NV12);
+            break;
+        }
+        default:
+            GAPI_Assert(false && "Unsupported media format for ONNX backend");
+    }
+    preprocess(roi.empty() ? pp : pp(roi), ti, dst);
```

Comment:
I'd put an assert here that `pp` is not empty.

Also: `pp` should remain in memory, otherwise it will be allocated every time to hold the color conversion result. It is costly for FHD and especially for 4K.

Reply:
...it means that `preprocess` should become a part of some object persistent with your ONNX RT session

Reply:
...the latter can be one more task (file in tracker and not address here)

Reply:
Added exMat in ONNXCompiled class.